File: src/programmers/level2/phone_num_list_42577/PhoneNumList.java
package programmers.level2.phone_num_list_42577;

import java.util.Arrays;

public class PhoneNumList {
    public boolean solution(String[] phone_book){
        Arrays.sort(phone_book);
        for (int i = 0; i < phone_book.length-1; i++) {
            if(phone_book[i+1].indexOf(phone_book[i]) == 0){
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        PhoneNumList pnl = new PhoneNumList();
        //test case
        String[][] phone_books = {{"119", "97674223", "1195524421"}, {"123","456","789"}, {"12","123","1235","567","88"}};
        for (String[] phone_book: phone_books) {
            System.out.println(pnl.solution(phone_book));
        }
    }
}

File: src/programmers/level2/phone_num_list_42577/phone_num_list.py
phone_book = ["123", "456", "789"]
def solution(phone_book):
    phone_book.sort()
    for i in range(0, len(phone_book)-1):
        if phone_book[i+1].find(phone_book[i]) == 0:
            return False
    return True

print(solution(phone_book))
